clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -906,6 +906,7 @@ public long position(TopicPartition partition, Duration timeout) {
                     return position.offset;
 
                 updateFetchPositions(timer);
+                timer.update();
```

Comment:
(this comment is unrelated to this PR)

It seems `AsyncConsumer#position` does not honour `WakupException`? see following test
```scala
  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersAll"))
  @Timeout(10)
  def testPositionRespectsWakeup(quorum: String, groupProtocol: String): Unit = {
    val topicPartition = new TopicPartition("abc", 15)
    val consumer = createConsumer()
    consumer.assign(List(topicPartition).asJava)
    val service = Executors.newSingleThreadExecutor()
    service.execute(() => {
      TimeUnit.SECONDS.sleep(1)
      consumer.wakeup()
    })
    try assertThrows(classOf[WakeupException], () => consumer.position(topicPartition, Duration.ofSeconds(3)))
    finally {
      service.shutdownNow()
      service.awaitTermination(1, TimeUnit.SECONDS)
    }
  }
```

Reply:
BTW, please let's me know if you have no free time. I'm fine to fix it if above bug I described is existent

Reply:
Thanks @chia7712 for catching that! Can you file a separate Jira to track? Thanks!

Reply:
https://issues.apache.org/jira/browse/KAFKA-16659